---
title: "Differential expression analysis of LeptospirillumFerriphilum using DeSeq2"
output: html_notebook
---

Load sample information into edgeR

```{r}
library(edgeR)
library(stringr)

count_csv <- read.csv("$HOME/Bioinformatics_data/lferriphilum/data/RNA_data/Salmon/final/LFerr_genome_assembly_plus_proteins_salmon_merged_rounded.csv", sep="\t", row.names="Name")
count_matrix <- as.matrix(count_csv)

cc <- rep('NULL', 15)
cc[c(1, 5, 15)] <- 'character'

annotation <- read.csv("$HOME/Bioinformatics_data/lferriphilum/analysis/DNA/04_genome_annotation/eggNOG-mapper/LFerr.ffn.emapper.annotations", sep="\t", colClasses=cc)

i = 2575
ann <- vector()
while (i > 0) {
  i <- i-1
  gn <- paste("PKGNLADN_", str_pad(i, 5, pad = "0"), sep="")
  if gn %in% annotation {
    idx <- match(gn, annotation)
    if annotation[idx][2] != "" {
      append(ann, annotation[idx][2])
    }
  } else {
    append(ann, gn)
  }
}

group <- c("Continuous", "Continuous", "Continuous", "Batch", "Batch")
design <- model.matrix( ~ group)

de <- DGEList(counts=count_matrix, group=group)

keep <- filterByExpr(de)
de <- de[keep, , keep.lib.sizes=F]

de <- calcNormFactors(de)
de_bak <- de
de
ann
```


Estimate dispersion (quantile-adjusted conditional maximum likelihood qCML)

```{r}

de <- estimateDisp(de)

```

BCV is the Viological Coefficient of Variation. The meaning is that the expression values vary up and down by BCV% between replicates.
Mathematically, BCV corresponds to the square root of the negative binomial dispersion for a gene across replicate samples. THis can be interpreted as the proportion of expression attributable to variability.

```{r}
logcpm <- cpm(de, log=TRUE)
plotMDS(logcpm)
plotBCV(de)
```

Test for differentially expressed genes

```{r}
et <- exactTest(de)
topTags(et)
```

```{r}
de <- estimateDisp(de, design, robust=T)
fit <- glmFit(de, design)
lrt <- glmLRT(fit)

topDE <- topTags(lrt, n=15L)
topDE
```



```{r}
plotMD(lrt)
```

```{r}
library(pheatmap)

# 01480 and 01549 are ribosomes and skew the whole plot, I'm gonna remove them
deList <- c("PKGNLADN_00968", "PKGNLADN_00970", "PKGNLADN_01008", "PKGNLADN_00463", "PKGNLADN_00244", "PKGNLADN_02086", "PKGNLADN_02392", "PKGNLADN_01009", "PKGNLADN_00243", "PKGNLADN_01508", "PKGNLADN_00966", "PKGNLADN_00245", "PKGNLADN_02088", "PKGNLADN_02386", "PKGNLADN_02530", "PKGNLADN_01427", "PKGNLADN_00775", "PKGNLADN_00969")


top <- count_matrix[topDE[,0],]
top_log <- log(top)
pheatmap(top_log, display_numbers = T)
top
top_log
```

23S ribosomal RNA
16S ribosomal RNA
APSA: Fumarate reductase succinate dehydrogenase flavoprotein domain protein
BMUL_1008: Metallophosphoesterase
hypothetical protein
FLIC: flagellin domain protein
Flagellar biosynthesis protein FlhF; eggNOG[FLHF]: GTP-binding signal recognition particle
Prokka: Cobalt-zinc-cadmium resistance protein CzcA; eggNOG[]: Heavy metal efflux pump, CzcA
PKGNLADN_00968 Prokka: Cation efflux system protein CusA; eggNOG[CUSA]: Heavy metal efflux pump, CzcA
hypothetical protein; eggNOG[]: Outer membrane efflux protein






