---
title: "Differential expression analysis of LeptospirillumFerriphilum using DeSeq2"
output: html_notebook
---

Load sample information into edgeR

```{r}
library(edgeR)
library(stringr)

count_csv <- read.csv("$HOME/Bioinformatics_data/lferriphilum/data/RNA_data/Salmon/final/LFerr_genome_assembly_plus_proteins_salmon_merged_renamed.tsv", sep="\t", row.names="Name")
count_matrix <- as.matrix(count_csv)

tpm_matrix  <- read.csv("$HOME/Bioinformatics_data/lferriphilum/data/RNA_data/Salmon/final/LFerr_genome_assembly_plus_proteins_salmon_merged_TPM_renamed.tsv", sep="\t", row.names="Name")

#group <- c("Batch", "Batch", "Continuous", "Continuous", "Continuous")
group <- c("Continuous", "Continuous", "Continuous", "Batch", "Batch")

de <- DGEList(counts=count_matrix, group=group)

design <- model.matrix( ~ group)

keep <- filterByExpr(de)
de <- de[keep, , keep.lib.sizes=F]

de <- calcNormFactors(de)
de_bak <- de
```


Estimate dispersion (quantile-adjusted conditional maximum likelihood qCML)

```{r}

de <- estimateDisp(de)

```

BCV is the Viological Coefficient of Variation. The meaning is that the expression values vary up and down by BCV% between replicates.
Mathematically, BCV corresponds to the square root of the negative binomial dispersion for a gene across replicate samples. THis can be interpreted as the proportion of expression attributable to variability.

```{r}
logcpm <- cpm(de, log=TRUE)
plotMDS(logcpm)
plotBCV(de)
```

Test for differentially expressed genes

```{r}
et <- exactTest(de)
topTags(et)
```

```{r}
de <- estimateDisp(de, design, robust=T)
fit <- glmFit(de, design)
lrt <- glmLRT(fit)

topDE <- topTags(lrt, n=17L)
topDE
```



```{r}
plotMD(lrt)
```

```{r}
library(pheatmap)

# 01480 and 01549 are ribosomes and skew the whole plot, I'm gonna remove them
rRNA <- list("PKGNLADN_01480", "PKGNLADN_01549")


topDE <- topDE[!(row.names(topDE) %in% rRNA), ]

topDE <- topDE[topDE != rRNA]

top <- tpm_matrix[row.names(topDE),]
top_log <- log(top)
pheatmap(top_log, display_numbers = T, cluster_cols=F)
#top
#top_log
```





